tests/end_to_end/end_to_end.cpp

```diff
@@ -492,8 +492,91 @@ bindmonitor_test(ebpf_execution_type_t execution_type)
 
     program_info_provider_t bind_program_info(EBPF_PROGRAM_TYPE_BIND);
 
-    result = ebpf_program_load(
-        SAMPLE_PATH "bindmonitor.o", nullptr, nullptr, execution_type, &object, &program_fd, &error_message);
+    const char* file_name = (execution_type == EBPF_EXECUTION_NATIVE ? "bindmonitor_um.dll" : "bindmonitor.o");
```

Comment:
Instead of modifying existing test, why not introduce a new test?

Reply:
Why? It's the same BPF program being tested, all that changes is execution type. This test can be run with:
1) Interpreter
2) JIT 
3) Native